lib/AcqList/hooks/useLocationFilters.js

```diff
@@ -80,6 +82,27 @@ const useLocationFilters = (location, history, resetData) => {
     [history, resetFilters],
   );
 
+  const clearLocationFilters = useCallback(
+    () => {
+      /*
+        there's a lot of state setters called synchronously both in this hook and it's consumers
+        so we need to be careful about not using old state data
+      */
+      setFilters((currentFilters) => {
+        const newFilters = pick(currentFilters, ['qindex', 'query']);
+
+        return newFilters;
+      }, (curState) => {
+        resetData();
+
+        history.push({
+          search: queryString.stringify(curState),
+        });
+      });
+    },
+    [history, setFilters, resetData],
```

Comment:
That's why we need to add it to the deps. Each time it changes `clearLocationFilters` should receive a new version.

Reply:
Only memorized values should be added to deps, otherwise, we will get the same without using `useCallback`.
> Each time it changes
In the current case, it triggers useCallback memorization even if nothing has changed in resetData.

Reply:
Also, this hook is used not only in Inventory, if we remove `resetData` from deps here but in other modules that function changes then we'll introduce bugs
I'd like to be safe and include all functions that are used in `useCallback` even if it means that we'll have some unnecessary memoizations